manager/assets/modext/widgets/element/modx.grid.tv.template.js

```diff
@@ -35,12 +35,74 @@ MODx.grid.TemplateVarTemplate = function(config) {
             ,width: 150
             ,sortable: true
         },{
+            header: _('category')
+            ,dataIndex: 'category_name'
+            ,width: 300
+        },{
             header: _('description')
             ,dataIndex: 'description'
             ,width: 300
         },tt]
+        ,tbar: ['->',{
+            xtype: 'modx-combo-category'
+            ,name: 'filter_category'
+            ,hiddenName: 'filter_category'
+            ,id: 'modx-tvtemp-filter-category'
+            ,emptyText: _('filter_by_category')
+            ,value: ''
+            ,allowBlank: true
+            ,width: 150
+            ,listeners: {
+                'select': {fn: this.filterByCategory, scope:this}
+            }
+        },'-',{
+            xtype: 'textfield'
+            ,name: 'query'
+            ,id: 'modx-tvtemp-search'
+            ,emptyText: _('search_ellipsis')
+            ,listeners: {
+                'change': {fn: this.search, scope: this}
+                ,'render': {fn: function(cmp) {
+                    new Ext.KeyMap(cmp.getEl(), {
+                        key: Ext.EventObject.ENTER
+                        ,fn: this.blur
+                        ,scope: cmp
+                    });
+                },scope:this}
+            }
+        },{
+            xtype: 'button'
+            ,id: 'modx-filter-clear'
+            ,text: _('filter_clear')
+            ,listeners: {
+                'click': {fn: this.clearFilter, scope: this}
+            }
+        }]
     });
     MODx.grid.TemplateVarTemplate.superclass.constructor.call(this,config);
 };
-Ext.extend(MODx.grid.TemplateVarTemplate,MODx.grid.Grid);
+Ext.extend(MODx.grid.TemplateVarTemplate,MODx.grid.Grid,{
+    filterByCategory: function(cb,rec,ri) {
+        this.getStore().baseParams['category'] = cb.getValue();
+        this.getBottomToolbar().changePage(1);
+        this.refresh();
```

Comment:
There's no need for calling `refresh()`, `changePage(1)` calls it, so you're calling it twice which is unnecessary and produces canceled request.